src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -3169,6 +3170,37 @@ .maxstack  2
   IL_0027:  ret
 }
 ");
+
+            var comp = (CSharpCompilation)verifier.Compilation;
+            var syntaxTree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var addressOfs = syntaxTree.GetRoot().DescendantNodes().OfType<PrefixUnaryExpressionSyntax>().ToArray();
+
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, addressOfs[0],
+                expectedSyntax: "&M",
+                expectedType: null,
+                expectedConvertedType: "delegate*<System.String, System.Void>",
+                expectedSymbol: "void C.M(System.String s)");
+
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, addressOfs[1],
+                expectedSyntax: "&M",
+                expectedType: null,
+                expectedConvertedType: "delegate*<out modreq(System.Runtime.InteropServices.OutAttribute) System.String, System.Void>",
+                expectedSymbol: "void C.M(out System.String s)");
+
+            string[] expectedMembers = new[] {
+                "void C.M(System.Object o)",
+                "void C.M(System.String s)",
+                "void C.M(out System.String s)",
+                "void C.M(System.Int32 i)"
+            };
+
+            AssertEx.Equal(expectedMembers, model.GetMemberGroup(addressOfs[0]).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
+            AssertEx.Equal(expectedMembers, model.GetMemberGroup(addressOfs[0].Operand).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
+
+            AssertEx.Equal(expectedMembers, model.GetMemberGroup(addressOfs[1]).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
+            AssertEx.Equal(expectedMembers, model.GetMemberGroup(addressOfs[1].Operand).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
         }
```

Comment:
>} [](start = 8, length = 1)

Is it interesting to test IOperation for this scenario, or is it covered elsewhere?

Reply:
I can add it, but I didn't consider to to be different in a meaningful sense from a regular address of as covered by the tests in `FunctionPointerOperations.cs` (since the real difference is in ensuring that `GetMemberGroup` behaves expectedly, and we don't have an IOperation equivalent for that).

---
In reply to: [430770434](https://github.com/dotnet/roslyn/pull/44436#discussion_r430770434) [](ancestors = 430770434)